Add marketing buttons to primer-marketing-buttons

gladwearefriends opened the pull request:
Testing on `github/github` here https://github.com/github/github/pull/81105 with the marketing buttons styles removed too.
Deployed onto review lab: https://testing-marketing-buttons-primer.review-lab.github.com/home

To test on review-lab, you can slap these classes onto one of the buttons on the homepage to test:
```
.btn-orange
.btn-outline-green
.btn-transparent
.btn-large
```

This has also been tested in StoryBook.

**This PR needs to be deployed after Primer 10 is released** https://github.com/github/github/pull/81106

---

This isn't in primer yet! https://github.com/github/github/blob/master/app/assets/stylesheets/custom/site/buttons.scss

Not sure if its related but i remembered this because I was checking out the style guide and the outline styles didnt load
https://styleguide.github.com/primer/components/marketing-buttons/

First time contributing 😬 Let me know if I did that right or not please!

/cc @primer/ds-core

gladwearefriends commented:
I'll open a PR to delete from github/github once I get the 👍

broccolini commented on modules/primer-marketing-buttons/README.md:
```diff
@@ -0,0 +1,91 @@
+# Primer Marketing CSS Typography
```
Looks like you copy+pasted the README for marketing Typography. The Docs for marketing buttons are mostly done and live in [github](https://github.com/github/github/tree/master/app/assets/stylesheets/primer-marketing/buttons) so you can copy that over, (long story but someone started creating a package but didn't actually add it to Primer lol), but not all the buttons you're adding here are in there, such as `btn-transparent` and `btn-blurple`—though I'd like to question the use of that button and/or a more sensible name before merging this in.

broccolini commented on modules/primer-marketing-buttons/README.md:
```diff
@@ -0,0 +1,91 @@
+# Primer Marketing CSS Typography
+
+[![npm version](http://img.shields.io/npm/v/primer-marketing-type.svg)](https://www.npmjs.org/package/primer-marketing-type)
```
You'll need to update these urls to `primer-marketing-buttons`.

broccolini commented on modules/primer-marketing-buttons/README.md:
```diff
@@ -0,0 +1,91 @@
+[![npm version](http://img.shields.io/npm/v/primer-marketing-type.svg)](https://www.npmjs.org/package/primer-marketing-type)
+[![Build Status](https://travis-ci.org/primer/primer-css.svg?branch=master)](https://travis-ci.org/primer/primer-css)
+
+> Flash messages, or alerts, inform users of successful or pending actions. Use them sparingly. Don’t show more than one at a time.
```
And this description should be for marketing buttons too... now going to check if this description is in marketing typography O_o

gladwearefriends replied:
so it _should_ be in marketing buttons? it was copied and pasted from marketing type. i just copy and pasted the message from primer-buttons instead.

broccolini commented on modules/primer-marketing-buttons/package.json:
```diff
@@ -0,0 +1,42 @@
+{
+  "version": "1.3.0",
```
The version number should be `0.0.1` until we're ready to ship.

broccolini commented on modules/primer-marketing-buttons/package.json:
```diff
@@ -0,0 +1,42 @@
+  "main": "build/index.js",
+  "primer": {
+    "category": "marketing",
+    "module_type": "utilities"
```
`module_type` should be `components`.

broccolini commented on modules/primer-marketing-buttons/package.json:
```diff
@@ -0,0 +1,42 @@
+    "lib",
+    "build"
+  ],
+  "repository": "https://github.com/primer/primer-css/tree/master/modules/primer-marketing-type",
```
You'll need to update the url to `primer-marketing-buttons`.

broccolini commented on modules/primer-marketing-buttons/package.json:
```diff
@@ -0,0 +1,42 @@
+{
+  "version": "1.3.0",
```
When this is ready to ship we can bump to major, but I think we should do that in the v10 pr.

broccolini commented on modules/primer-marketing-buttons/lib/button.scss:
```diff
@@ -0,0 +1,30 @@
+.btn-outline-green { @include btn-outline($green); }
+
+@mixin btn-transparent-active {
+  color: #333;
```
Forgot to add, can you replace the hex values with new color variables? We also have variables for line-height, border, and border-radius too (for use further down in `btn-transparent`.

gladwearefriends replied:
@broccolini will do! im actually going to remove a few declarations under`btn-transparent` on second thought a lot of these should be inherited from `.btn` already

broccolini commented on modules/primer-marketing-buttons/lib/button.scss:
```diff
@@ -0,0 +1,30 @@
+.btn-transparent {
+  padding: 0.55em 1em;
+  font-size: inherit;
+  font-weight: 500; // to avoid faux-bolding
```
I'm curious about what "faux-bolding" means, we've updated all our font-weights to bold (600), or normal (400) in most other places. If we do need it, you can use the `$font-weight-semibold` variable instead.

gladwearefriends replied:
i also just remembered a few of these styles from github/github where i copied them over from were literally declared specifically for where a `.btn-transparent` was used on the marketing homepage. ive removed the font-weight, font-size, padding, line-height, and border-radius as they should be inherited from `.btn` or the other btn size classes, e.g.`.btn-large`

gladwearefriends commented:
i made updates!

i also noticed `primer-marketing-support` isnt needed as a dependency for `primer-marketing-buttons` and took that out. and moved `.btn-large` from the `primer-buttons` module into `primer-marketing-buttons` as well. and updated readme to cover everything in the module. let me know how that looks!

broccolini commented on modules/primer-css/package.json:
```diff
@@ -42,6 +42,7 @@
     "primer-layout": "1.3.0",
     "primer-markdown": "3.6.0",
     "primer-marketing": "5.3.1",
+    "primer-marketing-buttons": "0.0.1",
```
testing

broccolini replied:
testing testing

gladwearefriends commented:
Testing on `github/github` here https://github.com/github/github/pull/81105 (tests failing due to the newly installed zuorajs)

gladwearefriends commented:
Related PR that needs to be deployed in `github/github` after Primer 10 is released https://github.com/github/github/pull/81106

This pr removes the styles that are now moved into the `primer-marketing` package under `primer-marketing-buttons`

broccolini commented on modules/primer-marketing-buttons/README.md:
```diff
@@ -0,0 +1,105 @@
+<a class="btn btn-outline-purple" href="#url" role="button">Link button</a>
+<button class="btn btn-outline-green" type="button">Button button</button>
+<div class="bg-gray-dark p-4 mt-4">
+  <button class="btn btn-transparent" type="button">
```
`btn-transparent` shouldn't be grouped with the outline options, it should have is own description and be a separate example.

broccolini commented on modules/primer-marketing-buttons/README.md:
````diff
@@ -0,0 +1,105 @@
+```
+
+## Additional sizing
+There are cases where you might want to increase the size of a button, for example when putting a main CTA inside of a jumbotron or major page callout. The `btn-large` class does the following to make a button more prominent:
````
I think the intro paragraph can be more concise, and I don't think we need to call out the description in a numbered list or get into exact values, suggest changing to something like the following:

> Use `btn-large` to increase the padding and border radius of a button. This is useful for prominent calls to action, and is often used in combination with [jumbtron](??) styles. 

> [Type scale utilities](typography/#type-scale-utilities) can be used to alter the font-size if needed. Padding is applied in em's so that it scales proportionally with the font-size.

**Side note:** As someone less familiar with marketing styles, I had no idea what a `jumbotron` might be. I looked it up and this looks like it could be called `hero` which I think is more meaningful to people. We try to stick with class names that are obvious, transparent, and human-readable—could we change this to something more meaningful in future? Related, `minitron` and `supertron` could just me `small` and `large` :woman_shrugging:. I get this is probably just a bit of humor, but it's un-helpful for people learning to use Primer. cc @sophshep

gladwearefriends replied:
I've changed the language in this section of the docs to "This is useful for prominent calls to action in hero sections."

Related, this section of the docs pertaining to `btn-large` is also moved into primer-buttons' readme.md as `btn-large`has been moved back to primer-buttons as well, where it originally was just without documentation.

broccolini commented on modules/primer-marketing-buttons/README.md:
````diff
@@ -0,0 +1,105 @@
+
+```html
+
+<p class="f3">
````
Show how the default `btn-large` looks without a font size change, then show another example with a type scale utility.

broccolini commented on modules/primer-marketing-buttons/lib/button.scss:
```diff
@@ -0,0 +1,32 @@
+  border-color: $white;
+}
+
+.btn-transparent {
```
This doesn't seem to be used anywhere in the `github/github` is it being used elsewhere for marketing sites?

gladwearefriends replied:
It is sometimes used on the ads we have on the homepage below the hero on github.com. I personally have also used it here on the hero and circular callouts https://partner.github.com/

jonrohan commented on modules/primer-marketing-buttons/package.json:
```diff
@@ -0,0 +1,41 @@
+    "test": "../../script/npm-run-all build lint test-docs"
+  },
+  "dependencies": {
+    "primer-support": "4.4.1"
```
I think `peerDependency` makes sense here. Be sure when you manually add it, that it's the current version of primer-buttons

broccolini replied:
I just had a flash back to trying to setup peer dependencies with Lerna from the original monorepo set up and it wasn't pretty. @gladwearefriends leave that to us, maybe it's all fine now since there's been lots of changes but I'd rather us deal with problems if they come up than you! We can do this before merging in tomorrow.

gladwearefriends commented:
To be included in the Changelog (as @broccolini requested above at https://github.com/primer/primer-css/pull/337#discussion_r148961326)

New module `primer-marketing-buttons`. Buttons for marketing websites at GitHub.

gladwearefriends commented on modules/primer-buttons/lib/button.scss:
```diff
@@ -88,13 +88,6 @@
 }
 
-// Large button adds more padding around text. Use font-size utils to increase font-size.. e.g, <p class="text-gamma"><button class="btn btn-large btn-primary" type="button">Big green button</button></p>
-.btn-large {
```
im moving `btn-large` back into primer core here! not sure why i moved it into marketing-buttons other than i thought it was only used on marketing pages 🤷🏻‍♀️

gladwearefriends commented on modules/primer-marketing-buttons/lib/button.scss:
```diff
@@ -0,0 +1,32 @@
+.btn-orange { @include btn-solid($white, lighten($orange-500, 7.5%), darken($orange-500, 7.5%)); }
+.btn-blurple { @include btn-solid($white, mix($blue-400, $purple-400, 35%), mix($blue-600, $purple-600, 35%)); }
```
These are good questions @broccolini. It looks like theyre only used on the business pages here: https://github.com/business and on `app/views/coupons/business_plus_only/show.html.erb` (not sure where that view is). I'm for keeping this as a one off style in github/github. im not sure if its used anywhere else on other one off marketing pages outside of github/github and it does seem weird to go out of the primer color system. What do you think @sophshep ?

At this point im gonna leave it out of this PR. I can put it back in if we think we should do it.
